lib/addressable/idna/native.rb

```diff
@@ -16,19 +16,11 @@
 #    limitations under the License.
 #++
 
-
+# libidn1 implementing IDNA2003
 require "idn"
 
 module Addressable
   module IDNA
-    def self.punycode_encode(value)
-      IDN::Punycode.encode(value.to_s)
-    end
-
-     def self.punycode_decode(value)
-       IDN::Punycode.decode(value.to_s)
-     end
-
```

Comment:
This is not used anywhere, not tested and not publicly available in the `pure` implementation (it's `private`). It's also not available in `libidn2`. So I think we can safely remove it for consistency. And even if somebody somewhere was actually using this method manually for some reason (and it was working because they used `libidn`), they can easily replace the one line call with the `IDN::Punycode` directly.

Reply:
Yes, agree.